katpoint/pointing.py

```diff
@@ -320,9 +334,17 @@ def fit(self, az, el, delta_az, delta_el, sigma_daz=None, sigma_del=None, enable
         assert az.shape == el.shape == delta_az.shape == delta_el.shape == sigma_daz.shape == sigma_del.shape, \
             'Input parameters should all have the same shape'
 
-        # Blank out the existing model
-        self.set()
+        if not keep_disabled_params:
+            # Blank out the existing model but warn that this behaviour is deprecated
+            self.set()
+            warnings.warn('Pointing model parameters that are not being fitted will be kept in '
+                          'future and not zeroed - zero the model beforehand instead', FutureWarning)
+        param_vector = np.array(self.values())
         sigma_params = np.zeros(len(self))
+        # Subtract the existing model from data (both enabled and disabled parameters)
+        fixed_delta_az, fixed_delta_el = self.offset(az, el)
+        residual_delta_az = delta_az - fixed_delta_az
+        residual_delta_el = delta_el - fixed_delta_el
```

Comment:
I am not entirely sure I understand this bit.   Is it getting the fixed_delta_xx from a offset pointing correction and when normally initialised this would be 0 ?

Reply:
The "old" model is considered to contain all the parameters that will be fixed during the fitting. We first have to subtract their contribution (the `self.offset(az, el)` part), and then fit the enabled parameters to the residual, and afterwards we add the fixed parameters back in again.

What I've realised is that we are effectively fitting a "delta" model instead of zeroing out the old model and starting fresh. There are three classes of parameters in this process:

- Parameters that are zero in the old model and not enabled will remain zero and be ignored.
- Parameters that are non-zero in the old model and not enabled with become fixed parameters (they keep their value after the fit).
- Parameters that are zero or non-zero in the old and enabled will be fit as before.

This way we don't have to identify the fixed parameters explicitly - we simply keep the old model around instead of zeroing it.